damus/Views/PostView.swift

```diff
@@ -609,38 +610,79 @@ struct PVImageCarouselView: View {
     var body: some View {
         ScrollView(.horizontal, showsIndicators: false) {
             HStack {
-                ForEach(media.map({$0.representingImage}), id: \.self) { image in
-                    ZStack(alignment: .topTrailing) {
-                        Image(uiImage: image)
-                            .resizable()
-                            .aspectRatio(contentMode: .fill)
-                            .frame(width: media.count == 1 ? deviceWidth*0.8 : 250, height: media.count == 1 ? 400 : 250)
-                            .cornerRadius(10)
-                            .padding()
-                            .contextMenu {
-                                if let uploadedURL = media.first(where: { $0.representingImage == image })?.uploadedURL {
-                                    Button(action: {
-                                        UIPasteboard.general.string = uploadedURL.absoluteString
-                                    }) {
-                                        Label(NSLocalizedString("Copy URL", comment: "Label for button in context menu to copy URL of the selected uploaded media asset."), image: "copy")
-                                    }
+                ForEach(media.indices, id: \.self) { index in
+                    ZStack(alignment: .topLeading) {
+                        if isVideo(url: media[index].uploadedURL) {
+                            VideoPlayer(player: configurePlayer(with: media[index].localURL))
+                                .frame(width: media.count == 1 ? deviceWidth * 0.8 : 250, height: media.count == 1 ? 400 : 250)
+                                .cornerRadius(10)
+                                .padding()
+                                .contextMenu { contextMenuContent(for: media[index]) }
+                        } else if is_animated_image(url: media[index].uploadedURL) {
+                            KFAnimatedImage(media[index].uploadedURL)
+                                .imageContext(.pfp, disable_animation: false)
```

Comment:
This does not seem right. This is not a profile picture, so we should set this to something else, probably `.note`

Reply:
That is correct. I have updated.